hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyPurgeResponse.java

```diff
@@ -69,6 +77,22 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
       }
     } else {
       processKeys(batchOperation, omMetadataManager);
+      processKeysToUpdate(batchOperation, omMetadataManager);
+    }
+  }
+
+  private void processKeysToUpdate(BatchOperation batchOp,
+      OMMetadataManager metadataManager) throws IOException {
+    if (keysToUpdateList == null) {
+      return;
+    }
+
+    for (SnapshotMoveKeyInfos keyToUpdate : keysToUpdateList) {
+      List<KeyInfo> keyInfosList = keyToUpdate.getKeyInfosList();
+      RepeatedOmKeyInfo repeatedOmKeyInfo =
+          createRepeatedOmKeyInfo(keyInfosList);
+      metadataManager.getDeletedTable().putWithBatch(batchOp,
```

Comment:
If this is put back to same deletedTable, keyDeletingService keeps on retrying ... do the above case of all version re-claimable -- not added to any list used to break this retry ?

Reply:
If all versions are reclaimable they are never put it `keysToModify`, they will be deleted by the `KeyDeletingService` itself. It won't even come here.